File: Leonidas/Final_Project/files/string_utils.py
# crypto_project/string_utils.py

def color_string(text, print_red=False):
    """
    Färbt einen Text auf der Konsole ein.
    Grün für normal, rot für Fehler / Attack.
    """
    bold_red = "\033[1;31m"
    green = "\033[32m"
    reset_color = "\033[0m"
    text_color = bold_red if print_red else green
    return text_color + str(text) + reset_color


def join_strings(items, delimiter):
    """
    Verbindet eine Liste von Strings mit einem Delimiter.
    Extra einfach geschrieben, so wie man es als Student machen würde.
    """
    result = ""
    for index, part in enumerate(items):
        result += str(part)
        if index < len(items) - 1:
            result += delimiter
    return result


def lower_string(s):
    """
    Liefert eine kleingeschriebene Version eines Strings zurück.
    """
    return str(s).lower()

File: Alexan/final_project/src/string_utils.cpp
#include "string_utils.h"
#include <cctype>     // Required for tolower()

std::string colorString(const std::string& str, const bool printRed)
{
    const std::string BOLD_RED = "\033[1;31m";
    const std::string GREEN = "\033[32m";
    const std::string RESET_COLOR = "\033[0m";
    const std::string TEXT_COLOR = (printRed) ? BOLD_RED : GREEN;
    return TEXT_COLOR + str + RESET_COLOR;
}

std::string join(const std::vector<std::string>& vec, const std::string& delimiter)
{
    std::string result;
    
    for(size_t n = 0; n < vec.size(); n++) {
        result += vec[n];

        if (n < vec.size() - 1) {
            result += delimiter;
        }
    }

    return result;
}

std::string lower(std::string str)
{
    for (char& c : str) {
        c = std::tolower(static_cast<unsigned char>(c));
    }
    return str;
}